newIDE/app/src/MainFrame/EditorTabs/EditorTabsHandler.js

```diff
@@ -289,3 +289,24 @@ export const getEventsFunctionsExtensionEditor = (
 
   return null;
 };
+
+export const moveTabToPosition = (
+  editorTabsState: EditorTabsState,
+  fromIndex: number,
+  toIndex: number
+): EditorTabsState => {
+  if (toIndex < fromIndex) toIndex += 1;
+  const currentEditorTabs = [...getEditors(editorTabsState)];
+  const movingTab = currentEditorTabs[fromIndex];
+  currentEditorTabs.splice(fromIndex, 1);
+  currentEditorTabs.splice(toIndex, 0, movingTab);
+
+  let currentTabIndex: number = getCurrentTabIndex(editorTabsState);
+  if (fromIndex === currentTabIndex) currentTabIndex = toIndex;
+  else if (fromIndex < currentTabIndex && toIndex >= currentTabIndex)
+    currentTabIndex -= 1;
+  else if (fromIndex > currentTabIndex && toIndex <= currentTabIndex)
+    currentTabIndex += 1;
+
+  return { editors: currentEditorTabs, currentTab: currentTabIndex };
```

Comment:
I think it's better to avoid to use the same variable for the condition and result.